src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationIconValueProvider.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.IO;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    // Sets the ApplicationIcon property to the path to the .ico file. This path is relative
+    // to the project file directory. If the selected file is not within the project directory
+    // tree, the file will be copied into the project file directory. Previously copied files
+    // are not deleted when a new file is selected. ApplicationIcon allows for both full paths
+    // and relative paths anywhere on disk, but we are not utilizing that in this implementation.
+    // The icon file does not need to be included in the project (such as a None/Content item).
+    [ExportInterceptingPropertyValueProvider(ApplicationIconPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class ApplicationIconValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private const string ApplicationIconPropertyName = "ApplicationIcon";
```

Comment:
We have constant `ConfigurationGeneral.ApplicationIconProperty` which should be used in this type instead of defining a new constant.

Reply:
Fixed.